libcusp/include/galois/graphs/MiningPartitioner.h

```diff
@@ -50,7 +50,7 @@ class MiningGraph : public DistGraph<NodeTy, EdgeTy> {
   std::unique_ptr<Partitioner> graphPartitioner;
 
   uint32_t G2LEdgeCut(uint64_t gid, uint32_t globalOffset) const {
-    assert(isLocal(gid));
+    assert(this->isLocal(gid));
```

Comment:
Is it more efficient?

Reply:
Not necessarily, but it's way more clear to a dev reading the code where the function is coming from.

Reply:
Make sense. Will change.

Reply:
It's not about performance. It's a stylistic choice. "this->" is redundant. The only reason you need it in this case is because the function is a member of the base class, so it is better to make that clear. Moreover, it is consistent with the rest of this class where "baseClass:": is used instead of "this->"

Reply:
Not really; without `this->`, `assert` is surprisingly not able to recognize the identifier `isLocal`. I'll change it to the base class along with other places used to use `this->` in the file.

Reply:
Idk. At least for me, the `this` is often helpful because it clarifies that a method is being called and not just some function somewhere. Whether or not to include the `this` pointer explicitly and whether or not to include the base class name doesn't strike me as something worth standardizing though.

Reply:
Agree with Ian; if the base class isn't used to prefix it like we're suggesting, then this-> should be used for the reason Ian wrote above (otherwise it's unknown where the function comes from, esp. if there are multiple headers).

Reply:
I used to think `this->` is the default but clang asked me to explicitly tell where the function comes from. For the sake of dev I'll prefix it with the base class.

Reply:
Either "baseClass::" or "this->" is required for correctness. @bozhiyou are you saying former is giving compilation errors?

Reply:
Either is fine. No prefix caused errors as the code used to be.